src/components/AttachmentModal.tsx

```diff
@@ -381,15 +381,22 @@ function AttachmentModal({
     /**
      * close the modal
      */
-    const closeModal = useCallback(() => {
-        setIsModalOpen(false);
+    const closeModal = useCallback(
+        (shouldCallDirectly?: boolean) => {
```

Comment:
Can you please explain **what** this new param does & why it's needed & why we only pass `true` from `onSelected`? 🙏

Reply:
> Can you please explain what this new param does

I add a new parameter to identify which cases require InteractionManager.runAfterInteractions and which do not, enabling the use of this parameter to handle turning it on and off.

> why it's needed & why we only pass true from onSelected

We don't add `InteractionManager.runAfterInteractions` for the selected `replace` option becasue:

1. There’s no need to add InteractionManager.runAfterInteractions because it doesn’t cause our bug in this case.
2. If we add InteractionManager.runAfterInteractions here and continue to navigate to another page, it will freeze the app. This is because the navigation happens first, and then the modal is dismissed, which causes the app to freeze. So, I added the true parameter to turn off InteractionManager.runAfterInteractions in closeModal